### PR TITLE
fix(release_server): fix how we merge values

### LINKED ISSUE #4337: Helm upgrade with reuse-values, without --set  gives YAML error
Output of `helm version`:
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:34:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
kops 1.9.1 on AWS

Example: https://github.com/nicdoye/helm-set-vs-reuse/tree/set-values-break-upgrades/helm/parent
(Note this is the `parent` chart in `set-values-break-upgrades` tag).

The parent chart requires the subchart, `subchart` conditionally on `subchart.enabled` equal `"true"`. 

Upgrading the parent *without* adding `--set subchart.enabled="true"` (or `"false"`) to the upgrade line fails.

```console
$ helm install helm/parent --set subchart.enabled="true" -n two-levels
NAME:   two-levels
...
$ helm upgrade two-levels helm/parent --reuse-values
Error: UPGRADE FAILED: yaml: line 4: could not find expected ':'
```

### PR DESCRIPTION
**Pull request by michelleN**
resolves #4337

### CODE REVIEW

**Comment by michelleN** [comment]
Working on a test for this so it's a wip.

**Comment by michelleN** [comment]
ready for review

**Comment by technosophos** [comment]
Hey licenseybot... plz rerun.

File: pkg/tiller/release_update_test.go [hunk 1]
```diff
@@ -129,6 +129,48 @@ func TestUpdateRelease_ResetValues(t *testing.T) {
...
+			Metadata: &chart.Metadata{Name: "hello"},
+			Templates: []*chart.Template{
+				{Name: "templates/hello", Data: []byte("hello: world")},
+				{Name: "templates/hooks", Data: []byte(manifestWithHook)},
+			},
+			Values: &chart.Config{Raw: "defaultFoo: defaultBar"},
+		},
+	}
+
+	fmt.Println("Running Install release with no overrides")
```
Comment (bacongobbler) [inline pkg/tiller/release_update_test.go]: would you mind removing this?

File: pkg/tiller/release_update_test.go [hunk 2]
```diff
@@ -129,6 +129,48 @@ func TestUpdateRelease_ResetValues(t *testing.T) {
...
+			Metadata: &chart.Metadata{Name: "hello"},
+			Templates: []*chart.Template{
+				{Name: "templates/hello", Data: []byte("hello: world")},
+			},
+		},
+		Values:      &chart.Config{Raw: ""},
+		ReuseValues: true,
+	}
+
+	fmt.Println("Running Update release with no override and reuse-values")
```
Comment (bacongobbler) [inline pkg/tiller/release_update_test.go]: would you mind removing this?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
two small comments, otherwise LGTM!

**Comment by bacongobbler** [comment]
tossing into 2.10 to fix a regression

**Comment by michelleN** [comment]
yea thanks let me fix those real quick

**Comment by michelleN** [comment]
@bacongobbler - ready for another look whenever you're ready

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
LGTM; merge whenever 🎉

File: pkg/tiller/release_server.go [hunk 3]
```diff
@@ -138,7 +138,10 @@ func (s *ReleaseServer) reuseValues(req *services.UpdateReleaseRequest, current
 		}
 
 		// merge new values with current
-		req.Values.Raw = current.Config.Raw + "\n" + req.Values.Raw
+		if current.Config != nil && current.Config.Raw != "" && current.Config.Raw != "{}\n" {
+			req.Values.Raw = current.Config.Raw + "\n" + req.Values.Raw
+		}
+		//req.Values.Raw = current.Config.Raw + "\n" + req.Values.Raw
```
Comment (technosophos) [inline pkg/tiller/release_server.go]: Can we remove the commented line?

**Pull request merged**